### PR TITLE
OptionsUI: support string list

### PR DESCRIPTION
**Pull request by ryantxu**
Support setting a list of strings in the options UI... this is needed for worldping

image

### CODE REVIEW

File: packages/grafana-data/src/utils/OptionsUIBuilders.ts [hunk 1]
```diff
@@ -143,6 +143,16 @@ export class PanelOptionsEditorBuilder<TOptions> extends OptionsUIRegistryBuilde
     });
   }
 
+  addStringArray<TSettings>(
```
Comment (dprokop) [inline packages/grafana-data/src/utils/OptionsUIBuilders.ts:146]: Haven't looked at the code yet - but I have doubts if we want this to be public API. This seems like quite a niche use case. Maybe you could just just `addCustomEditor` for world ping?

Comment (ryantxu) [reply packages/grafana-data/src/utils/OptionsUIBuilders.ts:146]: string[] niche?!   for field config sure... but isn't this for options in general?

Comment (dprokop) [reply packages/grafana-data/src/utils/OptionsUIBuilders.ts:146]: @ryantxu well, I've used wrong expression :) I was just wondering if we want to provide an UI fur this option, don't know how wildly used it is. But I'm totally fine with adding it- above is just wondering :)

File: packages/grafana-ui/src/components/OptionsUI/strings.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          );
+        })}
+
+        <Input
+          className={styles.textInput}
+          placeholder={item.settings?.placeholder || 'Enter text'}
+          defaultValue={''}
+          onBlur={e => this.onValueChange(e, -1)}
+          onKeyDown={e => this.onValueChange(e, -1)}
+          suffix={<Icon name="plus-circle" />}
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/OptionsUI/strings.tsx:75]: I would probably do `addonAfter` with a Button here instead.

File: packages/grafana-data/src/types/OptionsUIRegistryBuilder.ts [hunk 3]
```diff
@@ -72,6 +72,10 @@ export interface OptionsUIRegistryBuilderAPI<
   ): this;
 
   addTextInput?<TSettings extends StringFieldConfigSettings = StringFieldConfigSettings>(
+    config: OptionEditorConfig<TOptions, TSettings, string[]>
+  ): this;
+
+  addStringArray?<TSettings extends StringFieldConfigSettings = StringFieldConfigSettings>(
     config: OptionEditorConfig<TOptions, TSettings, string>
```
Comment (dprokop) [inline packages/grafana-data/src/types/OptionsUIRegistryBuilder.ts:79]: Think `string` and `string[]` are mixed here, aren't they?

File: packages/grafana-ui/src/components/OptionsUI/strings.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+import { Icon } from '../Icon/Icon';
+import { stylesFactory, getTheme } from '../../themes';
+import { css } from 'emotion';
+
+type Props = FieldConfigEditorProps<string[], StringFieldConfigSettings>;
+
+export class StringArrayEditor extends React.PureComponent<Props> {
+  constructor(props: Props) {
+    super(props);
+  }
```
Comment (dprokop) [inline packages/grafana-ui/src/components/OptionsUI/strings.tsx:13]:
```suggestion
```

Unnecessary constructor

File: packages/grafana-ui/src/components/OptionsUI/strings.tsx (hunk 2 above)
Comment (dprokop) [reply packages/grafana-ui/src/components/OptionsUI/strings.tsx:75]: I think I would follow the same pattern we use in i.e. data links:

image

We need to consult with @jessover9000 and @sarlinska as they are Guardians Of The UI :)

**Review by dprokop — APPROVED** [review]
Left nit that we need to talk through with the UX team. Other than that 👍

**Pull request merged**